src/papyrus-lang-vscode/src/features/commands/SubstitutionCommands.ts

```diff
@@ -0,0 +1,35 @@
+import { commands, Disposable } from 'vscode';
+import { IExtensionConfigProvider } from '../../ExtensionConfigProvider';
+import { take } from 'rxjs/operators';
+
+export class SubstitutionCommands implements Disposable {
+    private readonly _configProvider: IExtensionConfigProvider;
+    private _importsCommand: Disposable;
+    private _sourcesCommand: Disposable;
+
+    constructor(@IExtensionConfigProvider configProvider: IExtensionConfigProvider) {
+        this._configProvider = configProvider;
+        this.registerCommands();
+    }
+
+    private async registerCommands() {
+        const config = (await this._configProvider.config.pipe(take(1)).toPromise())['skyrimSpecialEdition'];
+
+        this._importsCommand = commands.registerCommand("papyrus.projectImports", () => {
+            return config.importDirs.join(';').replace('${installPath}', config.installPath);
+        });
+
+        this._sourcesCommand = commands.registerCommand("papyrus.projectSources", () => {
+            return config.sourceDirs.join(' ').replace('${installPath}', config.installPath);
+        });
+
+        this._sourcesCommand = commands.registerCommand("papyrus.projectSources0", () => {
```

Comment:
I'm unclear on what the `0` post-fixed command is.

Reply:
Also note that this means we leak a command by not disposing it since the original is lost after reassignment.